runtime-modules/content-directory/src/lib.rs

```diff
@@ -0,0 +1,1645 @@
+            for (id, new_value) in new_property_values.into_iter() {
+                // Try to find a current property value in the entity
+                // by matching its id to the id of a property with an updated value.
+                let current_prop_value = updated_values
+                    .get_mut(&id)
+                    // Throw an error if a property was not found on entity
+                    // by an in-class index of a property update.
+                    .ok_or(ERROR_UNKNOWN_ENTITY_PROP_ID)?;
+
+                // Skip update if new value is equal to the current one or class property type
+                // is locked for update from current actor
+                if new_value == *current_prop_value {
+                    continue;
+                } else {
+                    let (mut temp_entities_rc_to_increment_vec, mut temp_entities_rc_to_decrement_vec) = Self::perform_entity_property_value_update(&class, &entity, id, access_level, new_value, current_prop_value)?;
+                    entities_rc_to_increment_vec.append(&mut temp_entities_rc_to_increment_vec);
+                    entities_rc_to_decrement_vec.append(&mut temp_entities_rc_to_decrement_vec);
+                    updated = true;
+                }
+            }
+
+            // If property values should be updated:
+            if updated {
+
+                //
+                // == MUTATION SAFE ==
+                //
+
+                <EntityById<T>>::mutate(entity_id, |entity| {
+                    entity.values = updated_values;
+                });
+                entities_rc_to_increment_vec
+                    .iter()
+                    .for_each(|entities_rc_to_increment| {
+                        Self::increment_entities_rc(entities_rc_to_increment);
+                    });
+                entities_rc_to_decrement_vec
+                    .iter()
+                    .for_each(|entities_rc_to_decrement| {
+                        Self::decrement_entities_rc(entities_rc_to_decrement);
+                    });
+            }
+
+            Ok(())
+        }
+
+        pub fn clear_entity_property_vector(
+            origin,
+            actor: Actor<T>,
+            entity_id: T::EntityId,
+            in_class_schema_property_id: PropertyId
+        ) -> dispatch::Result {
+            let account_id = ensure_signed(origin)?;
+
+            Self::ensure_known_entity_id(entity_id)?;
+
+            let (entity, access_level) = Self::get_entity_and_access_level(account_id, entity_id, actor)?;
+
+
+            let current_property_value_vec =
+            Self::get_property_value_vec(&entity, in_class_schema_property_id)?;
+
+            Self::ensure_class_property_type_unlocked_for(
+                entity.class_id,
+                in_class_schema_property_id,
+                access_level,
+            )?;
+
+            let entities_rc_to_decrement = current_property_value_vec
```

Comment:
The mutation is already safe here, so move comment up.

Reply:
Done